src/Reflex/Spider/Internal.hs

```diff
@@ -935,17 +941,41 @@ type BehaviorEnv x = (Maybe (Weak (Invalidator x), IORef [SomeBehaviorSubscribed
 
 --type role BehaviorM representational
 -- BehaviorM can sample behaviors
-newtype BehaviorM x a = BehaviorM { unBehaviorM :: ReaderT (BehaviorEnv x) IO a } deriving (Functor, Applicative, MonadIO, MonadFix)
```

Comment:
What is wrong with `ReaderT`? The role of it's first parameter is representational. `BehaviorEnv`'s one parameter is phantom. I would think that GHC could thread the phantom fine?

Reply:
The trouble is the third parameter, which is nominal. :-(

Reply:
Ah right. Thanks.

Reply:
We could generalize to something like `RIO`, and probably should do so internally. As you suggest, there's no need to pin down the environment type too eagerly. Only the monad needs to be fixed.

Reply:
Why is the 3rd parameter nominal?  Is this a property of ReaderT?

Reply:
@ryantrinkle the third type parameter of `ReaderT` is nominal only because roles suck: the constraint one wants is `(Coerciable r0 r1, Coerciable (m0 a0) (m1 a1)) => Coerciable (ReaderT r0 m0 a0) (r1 m1 a1))`.